From 2978e899c6632e42d518068ef2a57576b0bf60cc Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 11 Oct 2024 14:19:26 +0200
Subject: [PATCH 1327/1695] dts: stm32: add SCMI HSI/HSE/MSI Kernel clocks for
 stm32mp21x

Management of kernel clocks for HSI, HSE, and MSI clocks.
These kernel clocks have their own gates.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0ce2d7886064b0ef494a35ad0481ceecab31cfa0 dts: stm32: add SCMI HSI/HSE/MSI Kernel clocks for stm32mp21x]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ie91d073f9cf1096f8460d56841f8d5943dc4c35b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/410786
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 42 +++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index f4dd1052b..8f90c8a77 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -945,6 +945,27 @@
 						domain-name = "ck_icn_m_etr";
 						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
 					};
+
+					clock@71 {
+						reg = <CK_SCMI_HSI_KER_CK>;
+						domain-name = "hsi_ker_ck";
+						clocks = <&rcc HSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@72 {
+						reg = <CK_SCMI_HSE_KER_CK>;
+						domain-name = "hse_ker_ck";
+						clocks = <&rcc HSE_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@73 {
+						reg = <CK_SCMI_MSI_KER_CK>;
+						domain-name = "msi_ker_ck";
+						clocks = <&rcc MSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 				};
 			};
 
@@ -1838,6 +1859,27 @@
 						domain-name = "ck_icn_m_etr";
 						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
 					};
+
+					clock@71 {
+						reg = <CK_SCMI_HSI_KER_CK>;
+						domain-name = "hsi_ker_ck";
+						clocks = <&rcc HSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@72 {
+						reg = <CK_SCMI_HSE_KER_CK>;
+						domain-name = "hse_ker_ck";
+						clocks = <&rcc HSE_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@73 {
+						reg = <CK_SCMI_MSI_KER_CK>;
+						domain-name = "msi_ker_ck";
+						clocks = <&rcc MSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 				};
 			};
 
-- 
2.47.2

